library/include/ck/library/reference_tensor_operation/cpu/reference_pooling_fwd.hpp

```diff
@@ -0,0 +1,343 @@
+// SPDX-License-Identifier: MIT
+// Copyright (c) 2018-2022, Advanced Micro Devices, Inc. All rights reserved.
+
+#pragma once
+
+#include <iostream>
+#include <sstream>
+#include <vector>
+#include <algorithm>
+
+#include "ck/tensor_operation/gpu/device/device_base.hpp"
+#include "ck/library/utility/host_tensor.hpp"
+#include "ck/library/utility/host_tensor_generator.hpp"
+
+namespace ck {
+namespace tensor_operation {
+namespace host {
+
+template <index_t InOutRank,
+          index_t WindowRank,
+          typename InDataType,
+          typename OutDataType,
+          typename AccDataType,
+          typename IndexDataType,
+          ck::ReduceTensorOp ReduceOpId,
+          bool PropagateNan,
+          bool OutputIndex>
+struct ReferencePoolingFwd : public device::BaseOperator
```

Comment:
Suggest to implement `ReferencePoolingFwd` to use `DevicePoolFwd` as its direct base class

Reply:
It is a good idea. But if I use the base class of DevicePoolFwd. I need to handle the descriptor.
As discussed in the teams, let's follow other reference deviceop, pass tensor directly for now.